internal/bootstrap/handlers/clients.go

```diff
@@ -49,28 +46,33 @@ func (_ *Clients) BootstrapHandler(
 
 	config := container.ConfigurationFrom(dic.Get)
 
-	var eventClient coredata.EventClient
-	var valueDescriptorClient coredata.ValueDescriptorClient
+	var eventClient interfaces.EventClient
 	var commandClient interfaces.CommandClient
-	var notificationsClient notifications.NotificationsClient
+	var notificationClient interfaces.NotificationClient
+	var subscriptionClient interfaces.SubscriptionClient
+	var deviceServiceClient interfaces.DeviceServiceClient
+	var deviceProfileClient interfaces.DeviceProfileClient
+	var deviceClient interfaces.DeviceClient
 
 	// Use of these client interfaces is optional, so they are not required to be configured. For instance if not
 	// sending commands, then don't need to have the Command client in the configuration.
 	if _, ok := config.Clients[clients.CoreDataServiceKey]; ok {
-		eventClient = coredata.NewEventClient(
-			local.New(config.Clients[clients.CoreDataServiceKey].Url() + clients.ApiEventRoute))
-
-		valueDescriptorClient = coredata.NewValueDescriptorClient(
-			local.New(config.Clients[clients.CoreDataServiceKey].Url() + clients.ApiValueDescriptorRoute))
+		eventClient = v2clients.NewEventClient(config.Clients[clients.CoreDataServiceKey].Url())
 	}
 
 	if _, ok := config.Clients[clients.CoreCommandServiceKey]; ok {
 		commandClient = v2clients.NewCommandClient(config.Clients[clients.CoreCommandServiceKey].Url())
 	}
 
+	if _, ok := config.Clients[clients.CoreMetaDataServiceKey]; ok {
+		deviceServiceClient = v2clients.NewDeviceServiceClient(config.Clients[clients.CoreMetaDataServiceKey].Url())
+		deviceProfileClient = v2clients.NewDeviceProfileClient(config.Clients[clients.CoreMetaDataServiceKey].Url())
+		deviceClient = v2clients.NewDeviceClient(config.Clients[clients.CoreMetaDataServiceKey].Url())
```

Comment:
simplify it with giving value from the map so don't need to access map so many times
```suggestion
	if val, ok := config.Clients[clients.CoreMetaDataServiceKey]; ok {
		deviceServiceClient = v2clients.NewDeviceServiceClient(val.Url())
		deviceProfileClient = v2clients.NewDeviceProfileClient(val.Url())
		deviceClient = v2clients.NewDeviceClient(val.Url())
```

Reply:
Yep, nice catch!